src/Compilers/CSharp/Portable/CommandLine/CSharpCompiler.cs

```diff
@@ -146,11 +145,29 @@ public override Compilation CreateCompilation(TextWriter consoleOutput, TouchedF
                 Arguments.CompilationOptions.
                     WithMetadataReferenceResolver(referenceDirectiveResolver).
                     WithAssemblyIdentityComparer(assemblyIdentityComparer).
-                    WithStrongNameProvider(strongNameProvider).
                     WithXmlReferenceResolver(xmlFileResolver).
                     WithSourceReferenceResolver(sourceFileResolver));
 
-            return compilation;
+            StrongNameProvider LoggingPortableStrongNameProvider()
+            {
+                return new PortableStrongNameProvider(Arguments.KeyFileSearchPaths)
+                {
+                    IOOp = new LoggingIOOperations(touchedFilesLogger)
+                };
+            }
+
+            StrongNameProvider LoggingDesktopStrongNameProvider()
+            {
+                return new DesktopStrongNameProvider(Arguments.KeyFileSearchPaths)
+                {
+                    IOOp = new LoggingIOOperations(touchedFilesLogger)
+                };
+            }
+
+            return compilation.WithOptions(
+                compilation.Feature("ByPassStrongName") == null ?
+                    compilation.Options.WithStrongNameProvider(LoggingPortableStrongNameProvider()) :
```

Comment:
>compilation.Options.WithStrongNameProvider [](start = 20, length = 42)

Why the local functions? why not just create the provider based on the switch and then pass it to `compilation.Options.WithStrongNameProvider`?

Reply:
Done.